File: app/src/main/java/com/bigscreen/mangindo/chapter/ChapterItemViewModel.java
package com.bigscreen.mangindo.chapter;


import android.content.res.Resources;

import com.bigscreen.mangindo.R;
import com.bigscreen.mangindo.network.model.Chapter;

public class ChapterItemViewModel {

    private final Resources resources;
    private final Chapter chapter;

    public ChapterItemViewModel(Resources resources, Chapter chapter) {
        this.resources = resources;
        this.chapter = chapter;
    }

    public String getChapterText() {
        if (isChapterNotNull()) {
            if (chapter.getTitle().contains("-")) {
                String[] titleArray = chapter.getTitle().split(" - ");
                if (titleArray.length == 2) {
                   return chapter.getTitle().split(" - ")[0];
                }
                return chapter.getTitle().replace("-", "");
            }
            return getFormattedChapterText(chapter.getHiddenChapter());
        }
        return "N/A";
    }

    public String getChapterTitle() {
        if (isChapterNotNull()) {
            if (chapter.getTitle().contains("-")) {
                String[] titleArray = chapter.getTitle().split(" - ");
                if (titleArray.length == 2) {
                    return chapter.getTitle().split(" - ")[1];
                }
                return "N/A";
            }
            return chapter.getTitle();
        }
        return "N/A";
    }

    private String getFormattedChapterText(String chapterNumber) {
        return String.format(resources.getString(R.string.chapter_), chapterNumber);
    }

    private boolean isChapterNotNull() {
        return chapter != null;
    }
}

File: app/src/main/java/com/bigscreen/mangindo/chapter/ChapterItemViewModel.kt
package com.bigscreen.mangindo.chapter

import android.content.res.Resources
import com.bigscreen.mangindo.R
import com.bigscreen.mangindo.network.model.Chapter


class ChapterItemViewModel(val resources: Resources, val chapter: Chapter?) {

    var chapterText: String = "N/A"
        get() {
            chapter?.let {
                if (it.title.contains("-")) {
                    val titleArray = it.title.split(" - ")
                    if (titleArray.size == 2) {
                        return it.title.split(" - ")[0]
                    }
                    return chapter.title.replace("-", "", true)
                }
                return getFormattedChapterText(it.hiddenChapter)
            }
            return "N/A"
        }

    var chapterTitle: String = "N/A"
        get() {
            chapter?.let {
                if (it.title.contains("-")) {
                    val titleArray = it.title.split(" - ")
                    if (titleArray.size == 2) {
                        return it.title.split(" - ")[1]
                    }
                    return "N/A"
                }
                return it.title
            }
            return "N/A";
        }

    private fun getFormattedChapterText(chapterNumber: String): String
            = String.format(resources.getString(R.string.chapter_), chapterNumber)

}